lib/spack/spack/hooks/sbang.py

```diff
@@ -157,6 +158,21 @@ def install_sbang():
     fs.set_install_permissions(sbang_bin_dir)
 
 
+def filter_shebangs_in_directory_and_warn(directory):
+    paths = filter_shebangs_in_directory(directory)
+
+    if not (paths and sbang_shebang_itself_too_long()):
+        return
+
+    short_list = elide_list(paths, 4)
+    sbang_bin = os.path.dirname(sbang_install_path())
+    tty.warn("Failed to shorten shebang lines of {0} files {1}, because sbang's "
+             "install path ({2}) is too long. For the installation to work, it is "
+             "required to have sbang in your PATH. Alternatively, you can shorten the "
+             "install root (config:install_tree:root).".format(
+                 len(paths), short_list, sbang_bin))
```

Comment:
This warning should mention that the packages will not be relocatable.

as long as you're modifying it, some newlines might make it easier for users to read also.